### ISSUE TITLE
.NET 7 support

### ISSUE THREAD

**Issue by SokoFromNZ** [dspace-group/dscom#91]
Heya,

I loove your dscom tool as it is the only feasable possibility for my project to get MSAL working on an old C++ app by calling current .NET Code from C++.

.NET 7 is out since yesterday and the released dscom32.exe throws an error when working on a .NET 7 Library.

Do you have any plans to support .NET 7 as quick as possible?

Thanks
Soko

**Comment by SOsterbrink** [comment]
Hi @SokoFromNZ, 

thank you for the compliments. It's always nice to hear from people who use the tools. 

The released dscom32.exe was built for .NET 6 and as far as I know loading .NET 7 files is expected to fail. If you want to load a .NET 7 file, you will have to build dscom for .NET 7 which you can try by cloning the repository, changing the <TargetFrameworks> and building it yourself. 
I personally haven't taken a look at .NET 7 yet, and don't know whether this would work to solve your problem. 

We may take a look at .NET in the coming weeks, but most likely we will only officially support the .NET LTS versions.  
https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core

**Comment by SokoFromNZ** [comment]
Hi,

Thanks for the info. A colleague of mine tried that (compiling with .NET 7) already but failed. He said the source code wanted to download something from github even though he downloaded your whole project as a ZIP.
Does your SLN have any ties to other github projects? Maybe he did something else wrong...

I suppose the only thing we have to change is the Targetframework in the CSPROJs and the thing should build, shouldn't it?

Thanks for your input
Soko

**Comment by SOsterbrink** [comment]
Yes, we use some NuGet packages like MinVer and XUnit in our projects. 
If these packages are not available for .NET 7, those are not found during the build and the build will fail. 

We use them, for example, in the [Test](https://github.com/dspace-group/dscom/blob/main/src/dscom.test/dscom.test.csproj) and in [Client](https://github.com/dspace-group/dscom/blob/main/src/dscom.client/dscom.client.csproj) 

If you load the solution in Visual Studio or Visual Studio Code with the Solution explorer, it will show you the project dependencies and consumed packages. 

Like I said, I haven't looked into .NET 7 yet and given that it was released yesterday I expect that many of the dependency projects also haven't looked at it yet.  
I expect that many packages will release a .NET 7 compatible version in the coming days which should allow you to change the `TargetFramework` for dscom but I cannot guarantee it.

**Comment by marklechtermann** [comment]
#92 

Fixed with Version v0.16.3 !

[https://github.com/dspace-group/dscom/releases/tag/v0.16.3](https://github.com/dspace-group/dscom/releases/tag/v0.16.3)

**Comment by SokoFromNZ** [comment]
@SOsterbrink 
I've had troubles including `net7.0` as Targetframework in the projects as I've got a NETSDK1045 error while building. Issue was your settings in global.json which basically freezed the SDK to .net6.

After solving this and finding your publish.bat - and chaning everything from net6 auf net7 in it - I still had the issue that this bat used the dlls in the net6 folder instead of the net7 (don't know why).

Ohh and I had to remove all your MinVer packages and references in your CSPROJ including the Github refs.

Anyhow... it did not work...

But thanks to @marklechtermann and his changes I don't have to deal with it :D

I've downloaded the EXEs v0.16.3 and it worked with .NET7 DLLs, awesome!!

Thanks a lot for the flash-like help
